src/main/webapp/js/script.js

```diff
@@ -0,0 +1,29 @@
+window.addEventListener('DOMContentLoaded', (event) => {
+  const cards = document.getElementsByClassName('card');
+  for (const card of cards) {
+    const up = card.getElementsByClassName('upvote')[0];
+    up.addEventListener('click', function() {
+      renderVote(card, up.value, 'upvote');
+    });
+    const down = card.getElementsByClassName('downvote')[0];
+    down.addEventListener('click', function() {
+      renderVote(card, down.value, 'downvote');
+    });
+  }
+});
+
+async function renderVote(card, url, action) {
+  const total = card.getElementsByClassName('total-votes')[0];
+  total.innerHTML = await updateVote(url, action);
+}
+
```

Comment:
i think you mentioned that you want to make t numbers longer than 3 digits to be display in a shorter way (1.0k)

Reply:
based on my searching, I don't think there's one clean way to do this and it will likely require some other js library. Probably better to make it a separate issue to be addressed in a future PR.